wordpress-mobile/WordPress-iOS
# Update PageListViewController deletion UI to match PostListViewController

Refs p2-pb3aDo-Ff#comment-1176. This PR updates the logic of the `PageListViewController` page to match that of the posts list:

* Deletion / Trash options in the ... menu of posts now uses the destructive style
* Tapping one of these options shows a confirmation alert before the action takes place
* I also removed the Publish Immediately option from the ... menu on the Trashed posts list, as it seemed odd to have it right next to the 'delete permanently' option, and it's not present on the posts list.

![trash-options](https://user-images.githubusercontent.com/4780/104230543-a0ab7580-5445-11eb-92be-e76e8da4ca58.png)

I considered pulling out a component that encapsulates this logic, or moving it to the AbstractPostListViewController to save on duplication (I took the method straight from PostListViewController) – however, I opted just to duplicated in this instance as a) It's only used in these two places, and b) The text in the UI needs to distinguish between Posts and Pages, so this seemed the simplest way.

**To test:**

* Build and run
* Navigate to the Pages screen for a site, and in either Published, Draft, or Scheduled, check that the option to Trash is in red and shows you a warning if you tap it.
* Check that if you cancel the warning nothing happens, and if you confirm it, the post is trashed.
* In the Trashed posts list, check that the Delete Permanently option in the ... menu is read, and the same logic as above applies to the confirmation alert.

PR submission checklist:
